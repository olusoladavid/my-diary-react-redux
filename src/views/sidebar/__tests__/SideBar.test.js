import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import { withRouter } from "react-router-dom";

import ConnectedSideBar, { SideBar } from "../SideBar.jsx";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const storeState = {
  user: { isAuthenticated: false }
};

const props1 = {
  isAuthenticated: false,
  location: { pathname: "/" },
  logOut: jest.fn()
};

const props2 = {
  isAuthenticated: true,
  location: { pathname: "/" },
  logOut: jest.fn()
};

it("should render without crashing when user is not authenticated", () => {
  const wrapper = shallow(<SideBar {...props1} />);
  expect(wrapper).toMatchSnapshot();
});

it("should render without crashing when user is authenticated", () => {
  const wrapper = shallow(<SideBar {...props2} />);
  expect(wrapper).toMatchSnapshot();
});

it("should change state when open is clicked", () => {
  const wrapper = shallow(<SideBar />);
  wrapper.instance().showSideBar();
  expect(wrapper.state("open")).toBe(true);
});

it("should change state when close is clicked", () => {
  const wrapper = shallow(<SideBar />);
  wrapper.instance().hideSideBar();
  expect(wrapper.state("open")).toBe(false);
});

it("should toggle sidebar", () => {
  const e = {
    target: {
      document: {
        documentElement: {
          clientWidth: 800
        }
      }
    }
  };
  const wrapper = shallow(<SideBar />);
  wrapper.instance().toggleSideBar(e);
  expect(wrapper.state("open")).toBe(true);
  e.target.document.documentElement.clientWidth = 600;
  wrapper.instance().toggleSideBar(e);
  expect(wrapper.state("open")).toBe(false);
});

it("should remove listener on component will mount", () => {
  const wrapper = shallow(<SideBar />);
  wrapper.instance().componentWillUnmount();
});

it("should synchronise with app state in the store", () => {
  const store = mockStore(storeState);
  const wrapper = shallow(<ConnectedSideBar />);
  const storeWrapper = shallow(wrapper.prop("children")({ ...props1, store }));
  expect(storeWrapper.prop("isAuthenticated")).toBe(false);
});
