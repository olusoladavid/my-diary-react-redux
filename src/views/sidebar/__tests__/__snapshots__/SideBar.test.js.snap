// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing when user is authenticated 1`] = `
<nav
  className="nav"
>
  <span
    className="nav__open-menu js-open-menu"
    onClick={[Function]}
  />
  <ul
    className="list nav__list js-nav-list"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <li
      className="list__item"
    >
      <span
        className="nav__close-menu js-close-menu"
        onClick={[Function]}
      />
    </li>
    <li
      className="list__item"
    >
      <span
        className="link nav__link"
        onClick={[MockFunction]}
      >
        Logout
      </span>
    </li>
  </ul>
</nav>
`;

exports[`should render without crashing when user is not authenticated 1`] = `
<nav
  className="nav"
>
  <span
    className="nav__open-menu js-open-menu"
    onClick={[Function]}
  />
  <ul
    className="list nav__list js-nav-list"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <li
      className="list__item"
    >
      <span
        className="nav__close-menu js-close-menu"
        onClick={[Function]}
      />
    </li>
    <li
      className="list__item"
      key="0"
    >
      <Link
        className="link nav__link"
        replace={false}
        to="/signup"
      >
        Signup
      </Link>
    </li>
  </ul>
</nav>
`;
